### PR TITLE
[Icon] Fix icon not having a default color to inherit

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

Fixes downstream regression that made the refactor to fill SVG using currentColor a breaking change. 

### WHAT is this pull request doing?

This PR adds `color` declaration to `.Icon svg` to inherit `currentColor` from in order to restore `Icon` having a default color. The prior default was the browser default `fill`, which is black and not aligned with the design language. Now the default Icon fill is `--p-color-icon`.

Added additional Icon stories to demo that this won't break the `currentColor` set up, or inheritance from components that render `Icon`.

🌀 [Spinstance](https://admin.web.test-default-icon-color.chloe-rice.us.spin.dev/store/shop1/dashboards)

| Before | After |
|--------|--------|
|Screenshot 2023-12-15 at 1 55 00 PM | Screenshot 2023-12-15 at 1 53 37 PM|

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Icon/Icon.module.scss [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 .Icon svg {
   fill: currentColor;
+  color: var(--p-color-icon);
```
Comment (aaronccasanova) [inline polaris-react/src/components/Icon/Icon.module.scss:14]: Should we instead set the default `tone` to `base`? Structuring our styles like this is what caused our specificity issues during the Uplift e.g.

```scss
.root { // Specificity 10
  color: red;
  #{$se23} & { // Specificity 21
    color: green;
  }
}

.primary { // Specificity 10 
  color: blue;
}
```

```jsx
<MyComponent variant="primary" /> // Renders `color: green`
```

Comment (chloerice) [reply polaris-react/src/components/Icon/Icon.module.scss:14]: I was considering that route, but it felt like a breaking change. Once CI is done, I'll make a temp commit exploring that to see if  there's any blocking edge cases 👀 🙏🏽

**Pull request closed without merging**